### PR TITLE
feat: add localization score adapter

### PR DESCRIPTION
**Pull request by TakumiKozaka-T4**
## PR Type

- New Feature

## Related Links

Need to add [new message](https://github.com/tier4/tier4_autoware_msgs/pull/42)
Need to fix [launcher](https://github.com/tier4/autoware_launch/pull/404)

## Description

Provide standard API of localization score for external applications.

## Review Procedure

1. Download sample data for rosbag replay simmulation. [Link](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/)
2. Run logging simulator.
3. Open another terminal and replay the rosbag.
4. Open another terminal and run the following command. `ros2 launch autoware_api_launch external_api_adaptor.launch.py 
`
5. Check the output by the command `ros2 topic echo /api/external/get/localization_scores`

The result will be displayed as follows.

> values:
> - name: transform_probability
>   value: 5.909358501434326
> - name: nearest_voxel_transformation_likelihood
>   value: 3.06380295753479

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s).
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: autoware_iv_external_api_adaptor/src/localization_score.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+void LocalizationScore::callbackNvtlScore(const Float32Stamped::ConstSharedPtr msg_ptr)
+{
+  is_nvtl_received_ = true;
+  score_nvtl_.value = msg_ptr->data;
+}
+
+void LocalizationScore::callbackTimer()
+{
+  LocalizationScoreArray localizatoin_scores_msg;
```
Comment (isamu-takagi) [inline autoware_iv_external_api_adaptor/src/localization_score.cpp]:
```suggestion
  LocalizationScoreArray localization_scores_msg;
```

File: autoware_iv_external_api_adaptor/src/localization_score.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    std::bind(&LocalizationScore::callbackTpScore, this, _1));
+  sub_nearest_voxel_transformation_likelihood_ = this->create_subscription<Float32Stamped>(
+    "/localization/pose_estimator/nearest_voxel_transformation_likelihood", 1,
+    std::bind(&LocalizationScore::callbackNvtlScore, this, _1));
+
+  // Timer callback
+  auto timer_callback = std::bind(&LocalizationScore::callbackTimer, this);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(1.0 / status_pub_hz_));
+  // timer_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_callback)>>(
```
Comment (isamu-takagi) [inline autoware_iv_external_api_adaptor/src/localization_score.cpp]: Please remove the commented code.

Comment (TakumiKozaka-T4) [reply autoware_iv_external_api_adaptor/src/localization_score.cpp]: Removed.

File: autoware_iv_external_api_adaptor/src/localization_score.cpp (hunk 1 above)
Comment (TakumiKozaka-T4) [reply autoware_iv_external_api_adaptor/src/localization_score.cpp]: Fixed.

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Pull request merged**